diff --git a/node_modules/react-native-vision-camera-worklets/android/CMakeLists.txt b/node_modules/react-native-vision-camera-worklets/android/CMakeLists.txt
index a569fda..b137096 100644
--- a/node_modules/react-native-vision-camera-worklets/android/CMakeLists.txt
+++ b/node_modules/react-native-vision-camera-worklets/android/CMakeLists.txt
@@ -18,6 +18,7 @@ include(${CMAKE_SOURCE_DIR}/../nitrogen/generated/android/VisionCameraWorklets+a
 include_directories(
         "src/main/cpp"
         "../cpp"
+        "../../react-native-worklets/Common/cpp"
 )
 
 find_library(LOG_LIB log)
diff --git a/node_modules/react-native-vision-camera-worklets/cpp/JSIConverter+AsyncQueue.hpp b/node_modules/react-native-vision-camera-worklets/cpp/JSIConverter+AsyncQueue.hpp
index 5b93f23..a6e2ce2 100644
--- a/node_modules/react-native-vision-camera-worklets/cpp/JSIConverter+AsyncQueue.hpp
+++ b/node_modules/react-native-vision-camera-worklets/cpp/JSIConverter+AsyncQueue.hpp
@@ -8,7 +8,9 @@
 
 #include <NitroModules/JSIConverter.hpp>
 #include <jsi/jsi.h>
-#if __has_include(<worklets/RunLoop/AsyncQueue.h>)
+#if __has_include(<worklets/Public/AsyncQueue.h>)
+#include <worklets/Public/AsyncQueue.h>
+#elif __has_include(<worklets/RunLoop/AsyncQueue.h>)
 #include <worklets/RunLoop/AsyncQueue.h>
 #elif __has_include(<RNWorklets/worklets/RunLoop/AsyncQueue.h>)
 #include <RNWorklets/worklets/RunLoop/AsyncQueue.h>
diff --git a/node_modules/react-native-vision-camera-worklets/lib/createWorkletRuntimeForThread.js b/node_modules/react-native-vision-camera-worklets/lib/createWorkletRuntimeForThread.js
index 085729f..64ec062 100644
--- a/node_modules/react-native-vision-camera-worklets/lib/createWorkletRuntimeForThread.js
+++ b/node_modules/react-native-vision-camera-worklets/lib/createWorkletRuntimeForThread.js
@@ -10,6 +10,7 @@ import { HybridWorkletQueueFactory } from './internal/HybridWorkletQueueFactory'
  * on the Camera pipeline's thread without context switching.
  */
 export function createWorkletRuntimeForThread(thread) {
+    globalThis.__WORKLET_QUEUE_FACTORY = HybridWorkletQueueFactory;
     const workletQueue = HybridWorkletQueueFactory.wrapThreadInQueue(thread);
     return createWorkletRuntime({
         name: thread.id,
diff --git a/node_modules/react-native-vision-camera-worklets/src/createWorkletRuntimeForThread.ts b/node_modules/react-native-vision-camera-worklets/src/createWorkletRuntimeForThread.ts
index 82a306b..68aa0b5 100644
--- a/node_modules/react-native-vision-camera-worklets/src/createWorkletRuntimeForThread.ts
+++ b/node_modules/react-native-vision-camera-worklets/src/createWorkletRuntimeForThread.ts
@@ -17,6 +17,7 @@ import { HybridWorkletQueueFactory } from './internal/HybridWorkletQueueFactory'
 export function createWorkletRuntimeForThread(
   thread: NativeThread,
 ): WorkletRuntime {
+  (globalThis as any).__WORKLET_QUEUE_FACTORY = HybridWorkletQueueFactory
   const workletQueue = HybridWorkletQueueFactory.wrapThreadInQueue(thread)
 
   return createWorkletRuntime({
